Add --save and --save-dev Options to clib-install

darthtrevino opened the pull request:
Hey Team;
I've added the --save and --save-dev options to clib-install. I needed to update the dependencies with the latest version of parson to support pretty JSON writing. The gist of the update is in `src/clib-install.c`. My C is a little rusty, so feel free to throw any feedback my way.

Thanks!
Chris

Fixes #19
Fixes #20

darthtrevino commented:
Bah, my test is jacked. I'll post an update soon.

darthtrevino commented:
Because of how parson persists JSON - forward-slashes in repos come out escaped, which is a little ugly. `clib install` will still work though.
 
```json
{
    "name": "test-package",
    "dependencies": {
        "stephenmathieson\/tabs-to-spaces": "1.0.0"
    }
}
```

stephenmathieson commented on package.json:
```diff
@@ -16,7 +16,8 @@
     "littlstar/asprintf.c": "0.0.3",
-    "logger": "0.0.0"
+    "logger": "0.0.0",
+    "kgabis/parson": "master"
```
i'd much rather see an actual version pinned here, or at least a commit hash

stephenmathieson replied:
@darthtrevino i've forked the original `parson` repo and reverted the `\/` nonsense.  can you update to use [`clibs/parson@1.0.0`](https://github.com/clibs/parson/tree/1.0.0)?

darthtrevino replied:
Sure! I apologize for not getting back to this PR, life got busy after the
holiday break. I'll make some time this week to polish it up.

On Mon, Jan 11, 2016 at 9:33 AM, Stephen Mathieson <notifications@github.com
> wrote:

> In package.json
> <https://github.com/clibs/clib/pull/124#discussion_r49350564>:
>
> > @@ -16,7 +16,8 @@
> >      "strdup": "0.0.0",
> >      "Constellation/console-colors.c": "1.0.1",
> >      "littlstar/asprintf.c": "0.0.3",
> > -    "logger": "0.0.0"
> > +    "logger": "0.0.0",
> > +    "kgabis/parson": "master"
>
> @darthtrevino <https://github.com/darthtrevino> i've forked the original
> parson repo and reverted the \/ nonsense. can you update to use
> clibs/parson@1.0.0 <https://github.com/clibs/parson/tree/1.0.0>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/clibs/clib/pull/124/files#r49350564>.
>

stephenmathieson replied:
no worries!  it took me forever to review your initial commits haha

stephenmathieson commented on src/clib-install.c:
```diff
@@ -172,6 +186,52 @@ executable(clib_package_t *pkg) {
 
 #undef E_FORMAT
 
+char* concat(char *s1, char *s2) {
```
this is a bit awkward.  can we either offload this to a dependency?

stephenmathieson commented:
first, sorry for the _huge_ delay!

overall, this lgtm.  left a few comments inline though =)

darthtrevino commented:
Hey Steve, 
I've addressed your comments. One thought I have is about the --save and --save-dev shorthand. Presently -s is shorthand for --save and -e is shorthand for --save-dev. If we want this to act kind of like node, it may make more sense to use -S and -D respectively.

jwerle commented:
@darthtrevino Using `-S` and `-D` for shorthand makes sense to me!

darthtrevino commented:
Thanks @jwerle! I was extending the tests to check the shorthands and ran into a bug. I'll try to have a push up soon to fix it.

`clib install -d jwerle/fs.c`

```
"development": {
    "jwerle/fs": {
        "c": "0.1.1"
    }
}
```

jwerle commented:
@darthtrevino cool! yeah https://github.com/clibs/clib/commit/3397092e2be7a95b76f1a2ed06dda228e7939c36 looks good too

stephenmathieson commented:
awesome!  thanks!!

jb55 commented:
yay! :confetti_ball:
